Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/TranslatorMLKit.java

```diff
@@ -0,0 +1,151 @@
+import android.content.SharedPreferences;
+
+import java.util.Locale;
+import java.util.concurrent.CountDownLatch;
+
+import in.juspay.hyper.core.BridgeComponents;
+import kotlin.jvm.Synchronized;
+
+
+public class TranslatorMLKit {
+
+    public static String result;
+
+    public static Translator downloadModel(String destinationLanguage) {
+
+        TranslatorOptions options =
+                new TranslatorOptions.Builder()
+                        .setSourceLanguage("en")
+                        .setTargetLanguage(destinationLanguage.substring(0, 2).toLowerCase())
+                        .build();
+        final Translator translator =
+                Translation.getClient(options);
+
+
+
+        DownloadConditions conditions = new DownloadConditions.Builder()
+                .requireWifi()
+                .build();
+        translator.downloadModelIfNeeded(conditions)
+                .addOnSuccessListener(
+                        new OnSuccessListener<Void>() {
+                            @Override
+                            public void onSuccess(Void v) {
+                                Log.d("hello world", "download successful");
+                            }
+                        })
+                .addOnFailureListener(
+                        new OnFailureListener() {
+                            @Override
+                            public void onFailure(@NonNull Exception e) {
+                                Log.d("hello world", "download failed");
+
+                            }
+                        });
+
+        return translator;
+
+    }
+
+
+    public static void updateResult(String val){
+        result = val;
+    }
+
+    public static void translateString(String destinationAddress, String destinationLanguage, TextView textView){
+
+
+            Translator translator = downloadModel(destinationLanguage);
+
+            translator.translate(destinationAddress)
+                    .addOnSuccessListener(
+                            new OnSuccessListener<String>() {
+                                @Override
+                                public synchronized void onSuccess(String translatedText) {
+
+                                    updateResult(translatedText);
+
+                                    Animation fadeInAnimation = AnimationUtils.loadAnimation(textView.getContext(), R.anim.fadein);
+
+                                    textView.startAnimation(fadeInAnimation);
+                                    textView.setHeight(textView.getHeight());
+                                    textView.setText(translatedText);
+                                    Log.d("hello world", "now");
+
+
+                                }
+
+
+                            })
+                    .addOnFailureListener(
+                            new OnFailureListener() {
+                                @Override
+                                public void onFailure(@NonNull Exception e) {
+
```

Comment:
on failure case return the string as it is, without translating. Not required in case of translating and updating text view